aten/src/ATen/native/native_functions.yaml

```diff
@@ -11235,3 +11236,11 @@
 - func: unflatten_dense_tensors(Tensor flat, Tensor[] tensors) -> Tensor[]
   variants: function
   python_module: nn
+
+- func: nested_tensor_constructor(Tensor[] list, ScalarType dtype, Device device, bool pin_memory, bool channels_last) -> Tensor
+  variants: function
+
+- func: is_nt_impl(Tensor nt) -> bool
```

Comment:
This looks a bit questionable

Reply:
I wish there was a better way. We don't seem to have a generic function that checks whether a torch.Tensor is backed by a particular TensorImpl subclass. Writing ```is_nested_tensor_impl``` seemed verbose, but I'm happy to change the name.

Reply:
So, if we were to do this completely analogously to sparse, there wouldn't be a native_functions.yaml entry, and various implementations of is_nested would be directly implemented in relevant areas. TBH, the way these are currently added is kind of crappy, but if it's crappy in the same way as is_sparse/is_sparse_csr it will be easier for us to fix it later.